src/ui/pages/payment-entry-page.svelte

```diff
@@ -150,27 +203,38 @@
   };
 </script>
 
+{#if emailError}
+  <ErrorPage
+    lastError={emailError}
+    {productDetails}
+    supportEmail={brandingInfo?.support_email ?? null}
+    onContinue={handlePurchaseFlowError}
+  />
+{/if}
```

Comment:
It’s a bit odd that we have different ways of representing `emailError`s and `modalErrorMessage`s. Is there a possibility to unify them?

Reply:
Yes I agree. But I am not sure I want to increase the scope of the PR and this is a pre-existing issue.

Reply:
I am actually having to come back to this since I am discovering some edge cases as I am adding more integration tests.